Source/Core/DolphinWX/LuaScripting.h

```diff
@@ -0,0 +1,101 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <map>
+#include <math.h>
+#include <mutex>
+#include <wx/artprov.h>
+#include <wx/bitmap.h>
+#include <wx/button.h>
+#include <wx/colour.h>
+#include <wx/filedlg.h>
+#include <wx/font.h>
+#include <wx/frame.h>
+#include <wx/gdicmn.h>
+#include <wx/icon.h>
+#include <wx/image.h>
+#include <wx/menu.h>
+#include <wx/settings.h>
+#include <wx/sizer.h>
+#include <wx/stattext.h>
+#include <wx/string.h>
+#include <wx/textctrl.h>
+#include "Core\HW\ProcessorInterface.h"
+#include "Core\State.h"
+#include "InputCommon\GCPadStatus.h"
+
+// Lua includes
+#ifdef __cplusplus
+#include <lua5.3\include\lua.hpp>
+#else
+#include <lua5.3\include\laux.h>
+#include <lua5.3\include\lua.h>
+#include <lua5.3\include\lualib.h>
+#endif
+
+typedef int (*LuaFunction)(lua_State* L);
+
+namespace Lua
+{
+extern std::map<const char*, LuaFunction>* registered_functions;
+
+// pad_status is shared between the window thread and the script executing thread.
+// so access to it must be mutex protected.
+extern GCPadStatus* pad_status;
+extern std::mutex lua_mutex;
+
+void ClearPad(GCPadStatus*);
+
+class LuaScriptFrame;
+
+class LuaThread : public wxThread
+{
+public:
+  LuaThread(LuaScriptFrame* p, wxString file);
+  ~LuaThread();
+
+  wxThread::ExitCode Entry();
+
+private:
+  LuaScriptFrame* parent = nullptr;
+  wxString file_path;
+};
+
+class LuaScriptFrame final : public wxFrame
+{
+public:
+  void Log(const char* message);
+  void GetValues(GCPadStatus* status);
+  void NullifyLuaThread();
+
+  LuaScriptFrame(wxWindow* parent);

```

Comment:
Constructor and destructor should appear first before other member functions in the class body.